tests/unit/test_model.py

```diff
@@ -428,6 +496,37 @@ def test_http_checksum_present(self):
             ["crc32", "crc32c", "sha256", "sha1"],
         )
 
+    def test_context_parameter_present(self):
+        service_model = model.ServiceModel(self.model)
+        operation = service_model.operation_model('ContextParamOperation')
+        self.assertEqual(len(operation.context_parameters), 1)
+        context_param = operation.context_parameters[0]
+        self.assertEqual(context_param.name, 'contextParamName')
+        self.assertEqual(context_param.member_name, 'ContextParamArg')
+
+    def test_context_parameter_absent(self):
+        service_model = model.ServiceModel(self.model)
+        operation = service_model.operation_model('OperationTwo')
+        self.assertIsInstance(operation.context_parameters, list)
+        self.assertEqual(len(operation.context_parameters), 0)
+
+    def test_static_context_parameter_present(self):
+        service_model = model.ServiceModel(self.model)
+        operation = service_model.operation_model('ContextParamOperation')
+        self.assertEqual(len(operation.static_context_parameters), 2)
+        static_ctx_param1 = operation.static_context_parameters[0]
+        static_ctx_param2 = operation.static_context_parameters[1]
+        self.assertEqual(static_ctx_param1.name, 'stringStaticContextParam')
+        self.assertEqual(static_ctx_param1.value, 'Static Context Param Value')
+        self.assertEqual(static_ctx_param2.name, 'booleanStaticContextParam')
+        self.assertEqual(static_ctx_param2.value, True)
+
+    def test_static_context_parameter_abent(self):
+        service_model = model.ServiceModel(self.model)
+        operation = service_model.operation_model('OperationTwo')
+        self.assertIsInstance(operation.static_context_parameters, list)
+        self.assertEqual(len(operation.static_context_parameters), 0)
```

Comment:
i think we should add some functional tests to make sure we can create a client with changes to the structure of service models

Reply:
I would argue that the test coverage you ask for already exists:

1. Any place that loads a service model should do so using `model.ServiceModel`. The unit tests in this PR already confirm that this class can handle service models with the new fields. (Note that there is no change in structure.)
2. Many functional test and all integration tests already require successful loading of a service model. I have already been running these tests locally with service model files that contain the new fields.